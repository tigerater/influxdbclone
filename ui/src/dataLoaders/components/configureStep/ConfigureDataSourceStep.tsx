// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import ConfigureDataSourceSwitcher from 'src/dataLoaders/components/configureStep/ConfigureDataSourceSwitcher'

// Types
import {DataLoaderStepProps} from 'src/dataLoaders/components/DataLoadersWizard'
import {DataLoaderType} from 'src/types/v2/dataLoaders'
import {Bucket} from '@influxdata/influx'

export interface OwnProps extends DataLoaderStepProps {
  type: DataLoaderType
  bucket: string
  org: string
  buckets: Bucket[]
}

type Props = OwnProps

@ErrorHandling
export class ConfigureDataSourceStep extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {type, bucket, org, buckets} = this.props

    return (
      <ConfigureDataSourceSwitcher
        buckets={buckets}
        bucket={bucket}
        org={org}
        dataLoaderType={type}
        onClickNext={this.handleNext}
      />
    )
  }

  private handleNext = async () => {
    const {onIncrementCurrentStepIndex, type, onExit} = this.props

    if (type === DataLoaderType.Scraping) {
      onExit()
      return
    }

    onIncrementCurrentStepIndex()
  }
}

export default ConfigureDataSourceStep
