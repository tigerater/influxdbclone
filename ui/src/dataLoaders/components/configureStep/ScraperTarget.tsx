// Libraries
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {
  Form,
  Input,
  Columns,
  Grid,
  ComponentSize,
  InputType,
  ComponentStatus,
} from 'src/clockface'
import BucketDropdown from 'src/dataLoaders/components/BucketsDropdown'
import {Bucket} from '@influxdata/influx'

interface Props {
  bucket: string
  buckets: Bucket[]
  onSelectBucket: (bucket: Bucket) => void
  onChangeURL: (value: string) => void
  onChangeName: (value: string) => void
  url: string
  name: string
}

export class ScraperTarget extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {onSelectBucket, url, name, bucket, buckets} = this.props
    return (
      <Grid>
        <Grid.Row>
          <Grid.Column widthXS={Columns.Eight} offsetXS={Columns.Two}>
            <Form.Element
              label="Name"
              errorMessage={this.nameEmpty && 'Target name is empty'}
            >
              <Input
                type={InputType.Text}
                value={name}
                onChange={this.handleChangeName}
                titleText="Name"
                size={ComponentSize.Medium}
                autoFocus={true}
                status={this.nameStatus}
              />
            </Form.Element>
            <Form.Element label="Bucket">
              <BucketDropdown
                selected={bucket}
                buckets={buckets}
                onSelectBucket={onSelectBucket}
              />
            </Form.Element>
          </Grid.Column>
          <Grid.Column widthXS={Columns.Eight} offsetXS={Columns.Two}>
            <Form.Element
              label="Target URL"
              errorMessage={this.urlEmpty && 'Target URL is empty'}
            >
              <Input
                type={InputType.Text}
                value={url}
                onChange={this.handleChangeURL}
                titleText="Target URL"
                size={ComponentSize.Medium}
                status={this.urlStatus}
              />
            </Form.Element>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    )
  }

  private get urlStatus(): ComponentStatus {
    if (this.urlEmpty) {
      return ComponentStatus.Error
    }
    return ComponentStatus.Default
  }

  private get urlEmpty(): boolean {
    return !this.props.url
  }

  private get nameStatus(): ComponentStatus {
    if (this.nameEmpty) {
      return ComponentStatus.Error
    }
    return ComponentStatus.Default
  }
  private get nameEmpty(): boolean {
    return !this.props.name
  }

  private handleChangeURL = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    this.props.onChangeURL(value)
  }

  private handleChangeName = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value
    this.props.onChangeName(value)
  }
}

export default ScraperTarget
