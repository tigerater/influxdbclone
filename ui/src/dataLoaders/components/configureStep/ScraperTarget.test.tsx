// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {ScraperTarget} from 'src/dataLoaders/components/configureStep/ScraperTarget'

const setup = (override = {}) => {
  const props = {
    bucket: '',
    buckets: [],
    onSelectBucket: jest.fn(),
    onChangeURL: jest.fn(),
    onChangeName: jest.fn(),
    url: '',
    name: '',
    ...override,
  }
  const wrapper = shallow(<ScraperTarget {...props} />)

  return {wrapper}
}

describe('ScraperTarget', () => {
  describe('rendering', () => {
    it('renders correctly with no bucket, url, name', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })

    it('renders correctly with bucket', () => {
      const {wrapper} = setup({bucket: 'defbuck'})
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })

    it('renders correctly with url', () => {
      const {wrapper} = setup({url: 'http://url.com'})
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })

    it('renders correctly with name', () => {
      const {wrapper} = setup({name: 'MyScraper'})
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
