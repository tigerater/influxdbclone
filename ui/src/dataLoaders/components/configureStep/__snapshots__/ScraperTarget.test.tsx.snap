// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScraperTarget rendering renders correctly with bucket 1`] = `
<Grid>
  <GridRow>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target name is empty"
        label="Name"
      >
        <Input
          autoFocus={true}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Name"
          type="text"
          value=""
        />
      </FormElement>
      <FormElement
        label="Bucket"
      >
        <BucketsDropdown
          buckets={Array []}
          onSelectBucket={[MockFunction]}
          selected="defbuck"
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target URL is empty"
        label="Target URL"
      >
        <Input
          autoFocus={false}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Target URL"
          type="text"
          value=""
        />
      </FormElement>
    </GridColumn>
  </GridRow>
</Grid>
`;

exports[`ScraperTarget rendering renders correctly with name 1`] = `
<Grid>
  <GridRow>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage={false}
        label="Name"
      >
        <Input
          autoFocus={true}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="default"
          titleText="Name"
          type="text"
          value="MyScraper"
        />
      </FormElement>
      <FormElement
        label="Bucket"
      >
        <BucketsDropdown
          buckets={Array []}
          onSelectBucket={[MockFunction]}
          selected=""
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target URL is empty"
        label="Target URL"
      >
        <Input
          autoFocus={false}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Target URL"
          type="text"
          value=""
        />
      </FormElement>
    </GridColumn>
  </GridRow>
</Grid>
`;

exports[`ScraperTarget rendering renders correctly with no bucket, url, name 1`] = `
<Grid>
  <GridRow>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target name is empty"
        label="Name"
      >
        <Input
          autoFocus={true}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Name"
          type="text"
          value=""
        />
      </FormElement>
      <FormElement
        label="Bucket"
      >
        <BucketsDropdown
          buckets={Array []}
          onSelectBucket={[MockFunction]}
          selected=""
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target URL is empty"
        label="Target URL"
      >
        <Input
          autoFocus={false}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Target URL"
          type="text"
          value=""
        />
      </FormElement>
    </GridColumn>
  </GridRow>
</Grid>
`;

exports[`ScraperTarget rendering renders correctly with url 1`] = `
<Grid>
  <GridRow>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage="Target name is empty"
        label="Name"
      >
        <Input
          autoFocus={true}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="error"
          titleText="Name"
          type="text"
          value=""
        />
      </FormElement>
      <FormElement
        label="Bucket"
      >
        <BucketsDropdown
          buckets={Array []}
          onSelectBucket={[MockFunction]}
          selected=""
        />
      </FormElement>
    </GridColumn>
    <GridColumn
      offsetXS={2}
      widthXS={8}
    >
      <FormElement
        errorMessage={false}
        label="Target URL"
      >
        <Input
          autoFocus={false}
          autocomplete="off"
          disabledTitleText="This input is disabled"
          name=""
          onChange={[Function]}
          placeholder=""
          size="md"
          spellCheck={false}
          status="default"
          titleText="Target URL"
          type="text"
          value="http://url.com"
        />
      </FormElement>
    </GridColumn>
  </GridRow>
</Grid>
`;
